### DISCUSSION TITLE
Bum version of Jackson to 2.12 to support Java records

### DISCUSSION THREAD

**Discussion by StanSvec** [dropwizard/dropwizard#4791] [Q&A] [answered]
Is it possible to migrate Jackson to version 2.12+ please? Jackson natively supports Java records since this version. At the moment `JsonProperty` annotation must be added for every record's field which is quite annoying.

**Comment by joschi** [comment, marked as answer]
@StanSvec Jackson 2.12.x is being used in Dropwizard 2.1.x.

There will not be an upgrade of Jackson 2.10.x to 2.12.x in Dropwizard 2.0.x in order to avoid breaking consumers in a patch release.

You can use the latest Dropwizard 2.1.x beta release if you need support for Java records out of the box.

You can also override the version of Jackson in your application build by importing the Jackson 2.12.x BOM.
I think it's compatible enough so that it will just work.

**Comment by StanSvec** [comment]
Thanks for making it clear @joschi. I'll close the ticket now.
